Index: python-crypto-2.0.1+dfsg1/Cipher/__init__.py
===================================================================
--- python-crypto-2.0.1+dfsg1.orig/Cipher/__init__.py	2009-03-20 19:10:41.000000000 +0100
+++ python-crypto-2.0.1+dfsg1/Cipher/__init__.py	2009-03-20 19:10:44.000000000 +0100
@@ -24,7 +24,7 @@
 """
 
 __all__ = ['AES', 'ARC2', 'ARC4',
-           'Blowfish', 'CAST', 'DES', 'DES3', 'IDEA', 'RC5',
+           'Blowfish', 'CAST', 'DES', 'DES3', 
            'XOR'
            ]
 
Index: python-crypto-2.0.1+dfsg1/Hash/__init__.py
===================================================================
--- python-crypto-2.0.1+dfsg1.orig/Hash/__init__.py	2009-03-20 19:10:41.000000000 +0100
+++ python-crypto-2.0.1+dfsg1/Hash/__init__.py	2009-03-20 19:10:44.000000000 +0100
@@ -19,6 +19,6 @@
 Crypto.Hash.SHA
 """
 
-__all__ = ['HMAC', 'MD2', 'MD4', 'MD5', 'RIPEMD', 'SHA', 'SHA256']
+__all__ = ['HMAC', 'MD2', 'MD4', 'MD5', 'SHA', 'SHA256']
 __revision__ = "$Id: __init__.py,v 1.6 2003/12/19 14:24:25 akuchling Exp $"
 
Index: python-crypto-2.0.1+dfsg1/Util/test.py
===================================================================
--- python-crypto-2.0.1+dfsg1.orig/Util/test.py	2009-03-20 19:10:41.000000000 +0100
+++ python-crypto-2.0.1+dfsg1/Util/test.py	2009-03-20 19:10:44.000000000 +0100
@@ -223,8 +223,7 @@
                         die('XOR failed on entry '+`entry`)
 
 
-def TestBlockModules(args=['aes', 'arc2', 'des', 'blowfish', 'cast', 'des3',
-                           'idea', 'rc5'],
+def TestBlockModules(args=['aes', 'arc2', 'des', 'blowfish', 'cast', 'des3'],
                      verbose=1):
     import string
     args=map(string.lower, args)
Index: python-crypto-2.0.1+dfsg1/setup.py
===================================================================
--- python-crypto-2.0.1+dfsg1.orig/setup.py	2009-03-20 19:10:44.000000000 +0100
+++ python-crypto-2.0.1+dfsg1/setup.py	2009-03-20 19:12:59.000000000 +0100
@@ -67,10 +67,6 @@
             Extension("Crypto.Hash.MD4",
                       include_dirs=['src/'],
                       sources=["src/MD4.c"]),
-            Extension("Crypto.Hash.RIPEMD",
-                      include_dirs=['src/'],
-                      sources=["src/RIPEMD.c"],
-                      libraries=HTONS_LIBS),
             Extension("Crypto.Hash.SHA256",
                       include_dirs=['src/'],
                       sources=["src/SHA256.c"]),
@@ -94,13 +90,6 @@
             Extension("Crypto.Cipher.DES3",
                       include_dirs=['src/'],
                       sources=["src/DES3.c"]),
-            Extension("Crypto.Cipher.IDEA",
-                      include_dirs=['src/'],
-                      sources=["src/IDEA.c"],
-                      libraries=HTONS_LIBS),
-            Extension("Crypto.Cipher.RC5",
-                      include_dirs=['src/'],
-                      sources=["src/RC5.c"]),
 
             # Stream ciphers
             Extension("Crypto.Cipher.ARC4",
